audformat/core/database.py

```diff
@@ -759,15 +820,8 @@ def join_field(
 
         # join tables
         for other in others:
-            # update misc tables first
-            # as they might be used in schemes
-            # linked by audformat tables
-            for misc_id, misc in other.misc_tables.items():
-                if misc_id in self.misc_tables:
-                    self[misc_id].update(misc, overwrite=overwrite)
-                else:
-                    self[misc_id] = misc.copy()
-            for table_id, table in other.tables.items():
+            for table_id in list(other.misc_tables) + list(other.tables):
+                table = other[table_id]
```

Comment:
To stay with `.items()` I did the following in `audb`, but I guess it is harder to understand and I should also change to your solution:
```python
for table_id, table in {**db.tables, **db.misc_tables}.items():
```

Reply:
I changed to 

```python
for table_id in list(db):
    table = db[table_id]
```
in `audb`.